### PR TITLE
Move NDC related code to ndc package

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Move NDC related code to ndc package

**Why?**
Refactor

**How did you test it?**
existing unit test and integration tests

**Potential risks**
cross DC replication fails

### CODE REVIEW

File: service/history/errors/errors.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package errors
```
Comment (yycptt) [inline service/history/errors/errors.go:21]: Let's check those error to see if they can be separated into the files where they are used.

File: service/history/execution/workflowExecutionUtil.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2017 Uber Technologies, Inc.
+// Copyright (c) 2020 Uber Technologies, Inc.
```
[the remark this answers is not in the document]
Comment (yux0) [reply service/history/execution/workflowExecutionUtil.go:1]: Done

File: service/history/reset/resetter.go [hunk 3]
```diff
@@ -23,10 +23,22 @@
 package reset
 
 import (
-	"context"
+	ctx "context"
+	"fmt"
+
+	"golang.org/x/net/context"
```
[the remark this answers is not in the document]
Comment (yux0) [reply service/history/reset/resetter.go:29]: Updated

File: service/history/workflowResetor_test.go [hunk 4]
```diff
@@ -25,6 +25,7 @@ package history
 import (
 	"context"
 	"fmt"
+
```
[the remark this answers is not in the document]
Comment (yux0) [reply service/history/workflowResetor_test.go:28]: Updated

File: service/history/constants/test_constants.go [hunk 5]
```diff
@@ -18,7 +18,7 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package testing
```
[the remark this answers is not in the document]
Comment (yux0) [reply service/history/constants/test_constants.go:21]: Updated

File: service/history/errors/errors.go (hunk 1 above)
Comment (yux0) [reply service/history/errors/errors.go]: Removed the errors package

File: service/history/constants/test_constants.go [hunk 6]
```diff
@@ -30,38 +30,38 @@ import (
 var (
 	// Version is the workflow version for test
 	Version = int64(1234)
```
Comment (yycptt) [inline service/history/constants/test_constants.go:32]: Also change this one?

**Comment by yycptt** [comment]
https://github.com/uber/cadence/issues/3181

**Pull request merged**